### PR TITLE
Provide a distiction of pre-release versions for non-master builds and release versions for master builds

### PR DESCRIPTION
**Pull request by PawelLipski**
Step towards https://github.com/VirtusLab/git-machete-intellij-plugin/issues/261, still missing the actual publish step

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,11 +70,47 @@ docker push gitmachete/intellij-plugin-ci
...
+
+Each PR must bump the version (see [version.gradle](version.gradle)) comparing to its base.
+
+Each regular (non-hotfix, non-release) PR is ultimately merged to `develop` and must have a non-empty pre-release version. <br/>
+Stacked PRs (Y -> X -> `develop`) are never merged until their base is finally changed to `develop`.
+They're must instead be retargeted to its base's base once their base branch is merged itself (Y -> X -> `develop` => X gets merged => Y -> `develop`).
+
+Each release PR (from `develop` to `master`) must not have a pre-release version.
+Once the release PR is merged, `master` is built. <br/>
+**TODO (#261): ** After a manual approval, the `master` build publishes the plugin to JetBrains marketplace.
```
Comment (PawelLipski) [inline CONTRIBUTING.md:109]: Done on another PR

File: version.gradle [hunk 2]
```diff
@@ -1,3 +1,3 @@
 ext {
-  PLUGIN_VERSION = '0.2.3'
+  PLUGIN_VERSION = '0.2.4-1'
```
Comment (mkondratek) [inline version.gradle:2]: :tada:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,3 +37,39 @@ To push the rebuilt image, you need write access to [`gitmachete` organization o
 DOCKER_BUILDKIT=1 BUILDKIT_PROGRESS=plain docker build -t gitmachete/intellij-plugin-ci .
 docker push gitmachete/intellij-plugin-ci
 ```
+
+
+## Versioning
```
Comment (mkondratek) [inline CONTRIBUTING.md:42]: :heart:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -18,14 +18,15 @@ ln -s ../../scripts/run-pre-build-checks .git/hooks/pre-commit
 Most non-standard/project-specific conventions are enforced by:
```
Comment (mkondratek) [inline CONTRIBUTING.md:18]: Line 43: you need **a** write access

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -36,3 +37,39 @@ To push the rebuilt image, you need write access to [`gitmachete` organization o
...
 ```
+
+
+## Versioning
+
+We follow [Semantic versioning](semver.org):
+
+* MAJOR version must be bumped for each plugin release that stops supporting any IDEA build (typically when `sinceBuild` is increased). <br/>
+  This does not apply to 0->1 major version transition, which is going to happen when the plugin's compatibility range is considered stable.
+* MINOR version must be bumped for each plugin release that either adds a new user-facing feature,
```
Comment (mkondratek) [inline CONTRIBUTING.md:48]: , :wink:

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -36,3 +37,39 @@ To push the rebuilt image, you need write access to [`gitmachete` organization o
...
+  Non-released plugin versions must **always** have a pre-release version indicator.
+
+
+## PRs & releases
+
+Each PR must bump the version (see [version.gradle](version.gradle)) comparing to its base.
+
+Each regular (non-hotfix, non-release) PR is ultimately merged to `develop` and must have a non-empty pre-release version. <br/>
+Stacked PRs (Y -> X -> `develop`) are never merged until their base is finally changed to `develop`.
+They're must instead be retargeted to its base's base once their base branch is merged itself (Y -> X -> `develop` => X gets merged => Y -> `develop`).
```
Comment (mkondratek) [inline CONTRIBUTING.md:63]: `They're must`?

Comment (PawelLipski) [reply CONTRIBUTING.md:63]: image

aka `moszcz (winny)`!

**Comment by PawelLipski** [comment]
Slightly reworked to fix develop builds, @mkondratek @micpiotrowski pls see https://github.com/VirtusLab/git-machete-intellij-plugin/pull/278/commits/87888969970d939d1732ee4b2503780960beb9e9
